### ISSUE TITLE
Improve CLI startup performance

### ISSUE THREAD

**Issue by auchenberg** [microsoft/vscode#66862]
It's been reported that starting VS Code via our CLI adds a significant delay to our startup performance. https://mobile.twitter.com/Snugug/status/1086289295088988161

A quick test on my Macbook Pro 13" 2018:
```
time code .
code .  0.17s user 0.14s system 48% cpu 0.641 total

time open "/Applications/Visual Studio Code.app"
open "/Applications/Visual Studio Code.app"  0.02s user 0.03s system 12% cpu 0.433 total
```

**Comment by hyperfekt** [comment]
Note that there is a significant delay after `code` returns but before a new window actually opens induced by using the CLI, therefore just using `time code` will not give an accurate picture.

**Comment by hyperfekt** [comment]
I pulled out my phone and made some measurements (all these are with vscode already open):
`code` returned after 692ms, the new window stole focus after 2283ms.
`code <file>` returned after 667ms, the window coming to the foreground stole focus after 1733ms.
Clicking on the menu entry for 'New Window` had the new window appearing after 149ms.
Opening a file through the file dialog opened by 'Open File...' had the new tab open after 33ms.

From that I conclude that there is a significant delay jts for using the CLI, most of which is not captured by `time` and which is larger (by more than just the cost of opening a new window itself) if a new window is opened.

(Given that I was neither in the mood to keep stepping through frames nor to code something to observe pixel changes, I did not make more than one run, but the results match my perception).
